lib/rules/jsx-newline.js

```diff
@@ -72,6 +72,22 @@ module.exports = {
     const jsxElementParents = new Set();
     const sourceCode = context.getSourceCode();
 
+    function isBlockCommentInCurlyBraces(element) {
+      const elementRawValue = sourceCode.getText(element);
+      return /^\s*{\/\*/.test(elementRawValue);
+    }
+
+    function findFirstNonBlockComment(index, elements) {
+      if (index >= elements.length) return;
+
+      const element = elements[index];
+      const isElementJSX = element.type === 'JSXElement' || element.type === 'JSXExpressionContainer';
+
+      if (!isBlockCommentInCurlyBraces(element) && isElementJSX) return element;
+
+      return findFirstNonBlockComment(index + 1, elements);
+    }
```

Comment:
is there a reason to use recursion here rather than `elements.find(predicate)`?

Reply:
No didn't think of using find. Let me give it a try.

Reply:
Added a version with `find` please let me no which version you prefer.